### ISSUE TITLE
ui5-input: aria-describedby content should be customizable

### ISSUE THREAD

**Issue by pankajpandey01** [SAP/ui5-webcomponents#5607]
### **Feature Request Description**
ui5-input: aria-describedby content is not customizable 
Related Problem:
We have a use case in shellbar search field where we need to provide some extra information to screen reader user about the search functionality. This is a descriptive information which can not be put as part of accessible-name (aria-label)

![Screen Shot 2022-08-01 at 4 01 54 PM](https://user-images.githubusercontent.com/67934622/182259630-c8c6d74a-c278-4565-8035-2a753532f228.png)

### **Proposed Solution**
ui5-input: aria-describedby content should be customizable

### **Priority**
- [ ] Low
- [x] Medium
- [ ] High
- [ ] Very High

A clear and concise description of the impact/urgency of the required feature.

### **Stakeholder Info** (if applicable)

- Organization: {Successfactors}

**Comment by MapTo0** [comment]
Hello @SAP/ui5-webcomponents-topic-rl 

As we spoke with @elenastoyanovaa I am forwarding that to the queue.

Regards,
Martin

**Comment by elenastoyanovaa** [comment]
Hello,

Currently there is no possibility to add descriptive text, due to the fact that the native input element would be in a different shadow DOM from the element which would be passed to the aria-describedby. You can find more information here in the Web Components & Accessibility section: https://sap.github.io/ui5-webcomponents/playground/advanced/accessibility/

In future we would like to expose an API similar to accessibleName called accessibleDescription and we will internally use aria-description. This attribute is part of the new W3C specification: https://w3c.github.io/aria/#aria-description, but this specification is currently a draft and the attribute is not yet supported in all major browsers: https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaDescription. This is the reason why we could not yet expose such an API.

Until then I could give you a suggestion for implementation. You could use our new Invisible Messaging service. The information about it could be found in the InvisibleMessaging section here https://sap.github.io/ui5-webcomponents/playground/advanced/accessibility/ There is also an example included. You could attach to the focus event of the input and call the announce method with the desired text.

Kind Regards,
Elena
